Add java examples to getting started guide, with language switch

raboof opened the pull request:
Needs a new publish from paradox (master) and akka-paradox (`groups` branch)

raboof committed:
WiP on Java version of new Hello World

Done:
- implementation for first 3 sections
- tests for first 2 sections

To do:
- formatting
- tests for section 3
- section 4 and 5
- hooking the fragments into the docs themselves

raboof committed:
Finish tutorial_03 conversion

raboof committed:
Converted tutorial 04

raboof committed:
Convert step 5 scala->java

raboof committed:
Reformat Java code (thx IntelliJ IDEA)

raboof committed:
Add java content to quickstart page

Example output at http://pi.bzzt.net/akka-docs-new/guide/quickstart.html

raboof committed:
Define paradox groups, use shorthand directives

raboof committed:
Use newer paradox snapshot

raboof committed:
Make tutorial apply to both java and scala

raboof committed:
Merge branch 'integrate-docs-scala-java' into language-switch

raboof committed:
Use sbt-paradox-akka

patriknw commented on akka-docs/src/main/paradox/scala/guide/quickstart.md:
```diff
@@ -111,14 +111,26 @@ The usual pattern is to have your system set up to stop on external signal (i.e.
+:   @@snip [HelloWorldApp.scala]($code$/scala/quickstart/HelloWorldApp.scala) { #full-example }
+
+Java
+:   @@snip [HelloWorldMain.scala]($code$/java/jdocs/quickstart/HelloWorldMain.java) { #full-example }
```
that thing inside the `[ ]` is not used/shown, but it should be .java

patriknw commented on akka-docs/src/main/paradox/scala/guide/tutorial_1.md:
```diff
@@ -208,14 +220,22 @@ represents the whole application. In other words, we will have a single top-leve
 The first actor happens to be rather simple now, as we have not implemented any of the components yet. What is new
-is that we have dropped using `println()` and instead use the `ActorLogging` helper trait which allows us to use the
-logging facility built into Akka directly. Furthermore, we are using a recommended creational pattern for actors; define a `props()` method in the [companion object](http://docs.scala-lang.org/tutorials/tour/singleton-objects.html#companions) of the actor:
+is that we have dropped using `println()` and instead use @scala[the `ActorLogging` helper trait] @java[`akka.event.Logging`] which allows us to use the
```
aaa, so that replaces the text, very nice!

patriknw commented on akka-docs/src/main/paradox/scala/guide/tutorial_1.md:
```diff
@@ -78,7 +78,11 @@ convenient terminology, and we will stick to it.
 
-@@snip [Hello.scala]($code$/scala/tutorial_1/ActorHierarchyExperiments.scala) { #print-refs }
+Scala
+:   @@snip [Hello.scala]($code$/scala/tutorial_1/ActorHierarchyExperiments.scala) { #print-refs }
```
why do we use Hello.scala / Hello.java for all snippets? what is the purpose of that part?

patriknw commented on akka-docs/src/main/paradox/scala/guide/tutorial_3.md:
```diff
@@ -100,48 +100,69 @@ message is preserved in the upper layers.* We will show you in the next section
+
+Java
+:   @@snip [Hello.java]($code$/java/jdocs/tutorial_3/DeviceGroupTest.java) { #device-group-test3 }
+>>>>>>> integrate-docs-scala-java:akka-docs-new/src/main/paradox/guide/tutorial_3.md
```
residue from merge conflict

patriknw commented on akka-docs/src/main/paradox/scala/guide/tutorial_3.md:
```diff
@@ -155,42 +176,59 @@ longer perform its duties after its collaborator actor has been stopped. In our
  been stopped:
 
-@@snip [Hello.scala]($code$/scala/tutorial_3/DeviceGroupSpec.scala) { #device-group-list-terminate-test }
+=======
```
what does that line mean? is that also from a merge?

patriknw commented on akka-docs/src/main/paradox/scala/guide/tutorial_4.md:
```diff
@@ -69,16 +73,19 @@ need to be able to work:
  * A timeout parameter, how long the query should wait for replies. Keeping this as a parameter will simplify testing.
 
+//getContext().getSystem().scheduler().scheduleOnce(
+        timeout, getSelf(), new CollectionTimeout(), getContext().dispatcher(), getSelf()
```
what's this?

raboof committed:
Fix snippet names

    find . -name "*.md" | xargs sed -i -e "s/snip \[[^]]*\]\(.*\)\/\([^\\\)]*\)/snip [\2]\1\/\2/"

raboof committed:
Remove leftover fragments

2m reviewed (approved):
LGTM. Its a bit pity that when redirecting from one language to another the position of the window is lost. We could do some additional magic, like storing Y offset in the browser before refreshing and then restoring after refresh. But that could be for later.

raboof commented:
@2m yes I agree it's a shame the location is lost, might indeed be interesting to see if we can come up with some trick, or otherwise it'll be fixed once we've merged the 2 trees together and can get rid for the refresh completely
